File: tmdb-core/src/main/java/dev/reuss/tmdb/common/image/LogoImage.java
package dev.reuss.tmdb.common.image;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Represents logo image metadata returned by TMDB.
 *
 * @param aspectRatio logo aspect ratio
 * @param filePath    TMDB image file path
 * @param height      logo height in pixels
 * @param id          TMDB image id
 * @param fileType    original logo file type, for example {@code .svg} or {@code .png}
 * @param voteAverage average image vote
 * @param voteCount   image vote count
 * @param width       logo width in pixels
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record LogoImage(
        @JsonProperty("aspect_ratio")
        double aspectRatio,

        @JsonProperty("file_path")
        String filePath,

        int height,

        String id,

        @JsonProperty("file_type")
        String fileType,

        @JsonProperty("vote_average")
        double voteAverage,

        @JsonProperty("vote_count")
        int voteCount,

        int width
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/common/image/LogoImage.kt
package dev.reuss.tmdb.common.image

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Represents logo image metadata returned by TMDB.
 *
 * @property aspectRatio logo aspect ratio
 * @property filePath TMDB image path for this image.
 * @property height      logo height in pixels
 * @property id          TMDB image id
 * @property fileType    original logo file type, for example {@code .svg} or {@code .png}
 * @property voteAverage Average user rating for the image reported by TMDB.
 * @property voteCount Number of user ratings for the image reported by TMDB.
 * @property width       logo width in pixels
 */
@JvmRecord
data class LogoImage(
    @all:JsonProperty("aspect_ratio")
    val aspectRatio: Double,
    @all:JsonProperty("file_path")
    val filePath: String?,
    @all:JsonProperty("height")
    val height: Int,
    @all:JsonProperty("id")
    val id: String?,
    @all:JsonProperty("file_type")
    val fileType: String?,
    @all:JsonProperty("vote_average")
    val voteAverage: Double,
    @all:JsonProperty("vote_count")
    val voteCount: Int,
    @all:JsonProperty("width")
    val width: Int,
) : TmdbModel
